manifester/helpers.py

```diff
@@ -21,3 +21,14 @@ def simple_retry(cmd, cmd_args=None, cmd_kwargs=None, max_timeout=240, _cur_time
         time.sleep(_cur_timeout)
         response = simple_retry(cmd, cmd_args, cmd_kwargs, max_timeout, new_wait)
     return response
+
+def process_sat_version(sat_version, valid_sat_versions):
+    if len(sat_version) > 8:
+        sat_version = sat_version.split('.')
+        sat_version = sat_version[:-1]
+        sat_version = ".".join(sat_version)
+    if sat_version not in valid_sat_versions:
+        sat_version = sat_version.split('.')
+        sat_version[1] = str(int(sat_version[1]) - 1)
+        sat_version = ".".join(sat_version)
```

Comment:
At this point, shouldn't you validate that the newly constructed sat version is valid?

Reply:
Good idea, I will add a check against the valid versions before returning `sat_version`.